File: javasample/src/main/java/com/github/stephenvinouze/advancedrecyclerview/javasample/adapters/SamplePaginationAdapter.java
package com.github.stephenvinouze.advancedrecyclerview.javasample.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.stephenvinouze.advancedrecyclerview.javasample.R;
import com.github.stephenvinouze.advancedrecyclerview.javasample.models.Sample;
import com.github.stephenvinouze.advancedrecyclerview.javasample.views.SampleItemView;
import com.github.stephenvinouze.advancedrecyclerview.pagination.adapters.RecyclerPaginationAdapter;

import org.jetbrains.annotations.NotNull;

/**
 * Created by stephenvinouze on 15/11/2017.
 */

public class SamplePaginationAdapter extends RecyclerPaginationAdapter<Sample> {

    @NotNull
    @Override
    protected View onCreateItemView(@NotNull ViewGroup parent, int viewType) {
        return new SampleItemView(parent.getContext());
    }

    @NotNull
    @Override
    protected View onCreateLoaderView(@NotNull ViewGroup parent, int viewType) {
        return LayoutInflater.from(parent.getContext()).inflate(R.layout.view_progress, parent, false);
    }

    @Override
    protected void onBindItemView(@NotNull View view, int position) {
        SampleItemView sampleItemView = (SampleItemView) view;
        sampleItemView.bind(getItems().get(position), isItemViewToggled(position));
    }

}

File: sample/src/main/java/com/github/stephenvinouze/advancedrecyclerview/sample/adapters/SamplePaginationAdapter.kt
package com.github.stephenvinouze.advancedrecyclerview.sample.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.github.stephenvinouze.advancedrecyclerview.pagination.adapters.RecyclerPaginationAdapter
import com.github.stephenvinouze.advancedrecyclerview.sample.R
import com.github.stephenvinouze.advancedrecyclerview.sample.models.Sample
import com.github.stephenvinouze.advancedrecyclerview.sample.views.SampleItemView

/**
 * Created by stephenvinouze on 14/11/2017.
 */
class SamplePaginationAdapter : RecyclerPaginationAdapter<Sample>() {

    override fun onCreateItemView(parent: ViewGroup, viewType: Int): View =
            SampleItemView(parent.context)

    override fun onBindItemView(view: View, position: Int) {
        when (view) {
            is SampleItemView -> view.bind(items[position], isItemViewToggled(position))
        }
    }

    override fun onCreateLoaderView(parent: ViewGroup, viewType: Int): View =
            LayoutInflater.from(parent.context).inflate(R.layout.view_progress, parent, false)

}